player/index.js

```diff
@@ -44,96 +37,108 @@ function parseWeekdayToInt(req, res, next, value) {
+      shiftStart = 0;
     } else if (dayPart === "p" || dayPart === "") {
-      req.shiftStart = 12;
+      shiftStart = 12;
     }
-    shiftStartSet = true;
-    next();
   } else {
     if (dayPart === "a" || dayPart === "") {
-      req.shiftStart = hour;
+      shiftStart = hour;
     } else if (dayPart === "p") {
-      req.shiftStart = hour + 12;
+      shiftStart = hour + 12;
     }
-    shiftStartSet = true;
-    next();
   }
 
-  if (!shiftStartSet) {
-    res.status(400).send(`'${value}' isn't a valid shift start value`);
+  if (typeof shiftStart === "undefined") {
+    throw new TypeError(`'${value}' isn't a valid shift start value`);
   }
-} 
 
-async function getFiles({ callSign, weekday, shiftStart } = params) {
+  return shiftStart;
+}
+
+async function getFiles({ callSign, weekday, shiftStart }) {
   const [files] = await storage.bucket(BUCKET).getFiles({
     prefix: `${callSign.toUpperCase()}-${weekday}-${shiftStart}`,
-  }); 
+  });
 
   // Return the two most recent recordings
   files.sort((a, b) => {
-    return a.metadata.timeCreated < b.metadata.timeCreated ? 1 : -1;
-  }); 
+    return a.name < b.name ? 1 : -1;
+  });
   return files.slice(0, 2);
 }
 
-function mapToLocals(files) {
-  return files.map(file => {
-    const d = new Date(file.metadata.timeCreated);
-    /* 
-      ensure that shifts that end at midnight 
-      are listed as their start date
-    */
-    d.setHours(d.getHours() - 2);
-    
-    const options = { 
-      weekday: "long", 
-      year: "numeric", 
-      month: "long", 
-      day: "numeric", 
-      timeZone: "America/Chicago" 
+function mapToLocals(files, shiftStart) {
+  return files.map((file) => {
+    const lastHyphen = file.name.lastIndexOf("-");
+    const isoDate = file.name.slice(lastHyphen - 10, lastHyphen);
+    const paddedHour = shiftStart.toString().padStart(2, "0");
+    const d = new Date(`${isoDate}T${paddedHour}:00:00-05:00`);
+
+    const options = {
+      weekday: "long",
+      year: "numeric",
+      month: "long",
+      day: "numeric",
+      timeZone: "America/Chicago",
     };
 
     return {
       title: d.toLocaleDateString("en-US", options),
-      src: file.metadata.mediaLink
-    }
-  }); 
+      src: file.metadata.mediaLink,
+    };
+  });
 }
 
-async function renderPlayer(req, res) {  
-  console.log(req.shiftStart);
-  const files = await getFiles({
-    callSign: req.params.callSign,
-    weekday: req.weekdayInt,
-    shiftStart: req.shiftStart,
-  });  
-  if(files.length === 0) {
-    res.sendStatus(404);
-  } else {
-    const streams = mapToLocals(files);    
-    res.render("player", { streams, layout: false });
-  }
+async function renderHtml(files, shiftStart) {
+  const streams = mapToLocals(files, shiftStart);
+  const template = await fs.readFile("./player.handlebars", "utf8");
+  const compiledTemplate = Handlebars.compile(template);
+  return compiledTemplate({ streams });
 }
 
+functions.http("player", async (req, res) => {
```

Comment:
The player is largely the same but has been updated to use the latest functions-framework and match the new signature for Cloud Functions. The new signature for HTTP functions provides an Express-like request and response context, so instead of returning a mini-Express app the function now uses those contexts to read parameters in from the path and render the same Handlebars template.